core/node/eigenda_proxy/src/lib.rs

```diff
@@ -5,20 +5,37 @@ use std::net::SocketAddr;
 
 use anyhow::Context as _;
 use axum::{
-    routing::{get, put},
+    extract::Path,
+    routing::{get, post},
     Router,
 };
+use eigenda_client::EigenDAClient;
+use request_processor::RequestProcessor;
 use tokio::sync::watch;
+use zksync_config::configs::da_client::eigen_da::EigenDAConfig;
+
 mod blob_info;
 mod eigenda_client;
 mod errors;
 mod memstore;
+mod request_processor;
 
-pub async fn run_server(mut stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
+pub async fn run_server(
+    config: EigenDAConfig,
+    mut stop_receiver: watch::Receiver<bool>,
+) -> anyhow::Result<()> {
     // TODO: Replace port for config
+
     let bind_address = SocketAddr::from(([0, 0, 0, 0], 4242));
     tracing::debug!("Starting eigenda proxy on {bind_address}");
-    let app = create_eigenda_proxy_router();
+
+    let eigenda_client = match config {
+        EigenDAConfig::Disperser(disperser_config) => EigenDAClient::new(disperser_config)
+            .await
+            .map_err(|e| anyhow::anyhow!("Failed to create EigenDA client: {:?}", e))?,
+        _ => panic!("memstore unimplemented"),
```

Comment:
Next would be implementing this for memstore also

Reply:
Yes! Working on it